src/transformers/models/recurrent_gemma/modeling_recurrent_gemma.py

```diff
@@ -252,7 +252,7 @@ def _update_cache(self, key_states, value_states, **cache_kwargs):
             to_shift = cache_position >= self.config.attention_window_size - 1
             indices = (slicing + to_shift[-1].int() - 1) % self.config.attention_window_size
 
-            k_out, v_out = self.key_states, self.value_states
+            k_out, v_out = self.key_states.to(key_states.device), self.value_states.to(value_states.device)
```

Comment:
Due to _setup_cache, self.key_states and self.value_states are initialized on the device of the hidden state that we pass to the model in generate (e.g. cuda:0). However, this layer might not be on the same device of the hidden state if we use multi-gpu. Hence, we need to make sure that `self.key_states` is on the same device as `key_states`. Same for `value_states`.